Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6378,7 +6378,13 @@ void HTMLMediaElement::visibilityStateChanged()
 
 bool HTMLMediaElement::requiresTextTrackRepresentation() const
 {
-    return (m_videoFullscreenMode != VideoFullscreenModeNone) && m_player ? m_player->requiresTextTrackRepresentation() : false;
+    if (m_videoFullscreenMode == VideoFullscreenModeNone || !m_player)
+        return false;
+#if PLATFORM(COCOA)
+    if (document().settings().blockMediaLayerRehostingInWebContentProcess())
+        return true;
```

Comment:
This seems wrong. We don't want to _always_ paint captions to an image and send to the UI process; only when we're in AVKit fullscreen or PiP mode.

Reply:
Removed, since this was not strictly needed.